sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/ApplicationInsightsRestClient.cs

```diff
@@ -46,16 +51,23 @@ internal async Task<int> InternalTrackAsync(ReadOnlyMemory<byte> body, Cancellat
             return message.TryGetProperty("ItemsAccepted", out var objItemsAccepted) && objItemsAccepted is int itemsAccepted ? itemsAccepted : 0;
         }
 
+        /// <summary>
+        /// Builds the ingestion Uri. (Example: https://dc.services.visualstudio.com/v2/track).
+        /// </summary>
+        internal RequestUriBuilder GetIngestionUri()
+        {
+            var uri = new RawRequestUriBuilder();
+            uri.AppendRaw(this.host, false);
+            uri.AppendPath($"/v{this.ApiVersion}/track", false);
+            return uri;
+        }
+
         internal HttpMessage CreateTrackRequest(IEnumerable<TelemetryItem> body)
         {
             var message = _pipeline.CreateMessage();
             var request = message.Request;
             request.Method = RequestMethod.Post;
-            var uri = new RawRequestUriBuilder();
-            uri.AppendRaw(host, false);
-            uri.AppendRaw("/v2", false);
-            uri.AppendPath("/track", false);
-            request.Uri = uri;
+            request.Uri = GetIngestionUri();
```

Comment:
This will not change frequently - consider optimizing it.

Reply:
Yes, i think this could be better optimized. To do that confidently, i would need to add some benchmark tests. I think this is out of scope of this specific PR